README.md

```diff
@@ -456,50 +456,91 @@ The following settings are optional and allow you to further configure your clus
+* `settings.kubernetes.eviction-soft`: The signals and thresholds that trigger pod eviction with a provided grace period.
+* `settings.kubernetes.eviction-soft-grace-period`: Delay for each signal to wait for pod termination before eviction.
 
-  Example user data for setting up eviction hard:
+  Example user data for setting up eviction values:
 
   ```toml
   [settings.kubernetes.eviction-hard]
   "memory.available" = "15%"
+
+  [settings.kubernetes.eviction-soft]
+  "memory.available" = "12%"
+
+  [settings.kubernetes.eviction-soft-grace-period]
+  "memory.available" = "30s"
+
+  [settings.kubernetes]
+  "eviction-max-grace-period" = 40
   ```
 
+  Remember to quote signals (since they all contain ".") and to quote all values.
+
 * `settings.kubernetes.image-gc-high-threshold-percent`: The percent of disk usage after which image garbage collection is always run, expressed as an integer from 0-100 inclusive.
 * `settings.kubernetes.image-gc-low-threshold-percent`: The percent of disk usage before which image garbage collection is never run, expressed as an integer from 0-100 inclusive.
 
-Since v1.14.0 `image-gc-high-threshold-percent` and `image-gc-low-threshold-percent` can be represented as numbers.
-For example:
+  Since v1.14.0 `image-gc-high-threshold-percent` and `image-gc-low-threshold-percent` can be represented as numbers.
+  For example:
 
-```toml
-[settings.kubernetes]
-image-gc-high-threshold-percent = 85
-image-gc-low-threshold-percent = 80
-```
+  ```toml
+  [settings.kubernetes]
+  image-gc-high-threshold-percent = 85
+  image-gc-low-threshold-percent = 80
+  ```
 
-For backward compatibility, both string and numeric representations are accepted since v1.14.0.
-Prior to v1.14.0 these needed to be represented as strings, for example:
+  For backward compatibility, both string and numeric representations are accepted since v1.14.0.
+  Prior to v1.14.0 these needed to be represented as strings, for example:
 
-```toml
-[settings.kubernetes]
-image-gc-high-threshold-percent = "85"
-image-gc-low-threshold-percent = "80"
-```
+  ```toml
+  [settings.kubernetes]
+  image-gc-high-threshold-percent = "85"
+  image-gc-low-threshold-percent = "80"
+  ```
 
-If you downgrade from v1.14.0 to an earlier version, and you have these values set as numbers, they will be converted to strings on downgrade.
+  If you downgrade from v1.14.0 to an earlier version, and you have these values set as numbers, they will be converted to strings on downgrade.
 
 * `settings.kubernetes.kube-api-burst`: The burst to allow while talking with kubernetes.
 * `settings.kubernetes.kube-api-qps`: The QPS to use while talking with kubernetes apiserver.
 * `settings.kubernetes.log-level`: Adjust the logging verbosity of the `kubelet` process.
   The default log level is 2, with higher numbers enabling more verbose logging.
+* `settings.kubernetes.memory-manager-policy`: The memory management policy to use: `None` (default) or `Static`.
```

Comment:
I double checked, but while `cpu-manager-policy` takes `none` and `static` (must be lower case), `memory-manager-policy` takes `None` and `Static` (must be title cased). So unfortunately there's an inconsistency, but it's an upstream inconsistency. Kubelet will fail to start if either of the settings are not cased correctly.

Reply:
> Let's move the documentation down closer to reserved-memory to group related settings together.

All current settings are in alphabetical order. I'd kind of hate to break that for one setting. What if I just add a "see `reserved-memory` for static memory manager configuration"?

Reply:
> What if I just add...

Oh, I did that with the next line. Let me know if that's not enough. For now I will also add a note to `reserved-memory` to reference back to this setting.

Reply:
One other idea if you don't think the cross referencing is enough. It might actually be nice to prepend `memory-manager-reserved-memory`. That would allow us to keep alphabetical sorting while keep these related settings together. It would also make it clear that the `memory-manager-reserved-memory` settings are related to the `memory-manager-policy`.

Just let me know what you think.

Reply:
> It might actually be nice to prepend memory-manager-reserved-memory. That would allow us to keep alphabetical sorting while keep these related settings together.

Yes, please - I like this idea.

Reply:
> I double checked, but while cpu-manager-policy takes none and static (must be lower case), memory-manager-policy takes None and Static (must be title cased). So unfortunately there's an inconsistency, but it's an upstream inconsistency. Kubelet will fail to start if either of the settings are not cased correctly.

We should be able to use [serde aliases](https://serde.rs/field-attrs.html#alias) to paper over the inconsistency:
```
 #[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, Scalar)]
 pub enum KubernetesMemoryManagerPolicy {
+    #[serde(alias = "static")]
     Static,
+    #[serde(alias = "none")]
     None,
 }
```

And perhaps give the CPU manager policy struct the same treatment. Then they will accept both forms (`"Static"` and `"static"`) when deserializing, but the expected form will be used for all serialization.

Reply:
`ValidCpuManagerPolicy` has `#[serde(rename_all = "lowercase")]` - that would be functionally equivalent, right?

Reply:
> `ValidCpuManagerPolicy` has `#[serde(rename_all = "lowercase")]` - that would be functionally equivalent, right?

That doesn't quite do the same thing - it means serde will expect the lowercase form when deserializing, and serialize into the lowercase form. Which is correct for the CPU manager policy, but not the memory manager policy.

What we want is to always serialize / deserialize into the expected form, but *also* to accept certain aliases when deserializing. So we'd need to add `Static` and `None` aliases on `ValidCpuManagerPolicy` variants (it already accepts / expects "static" and "none").